### DISCUSSION TITLE
More variables by default or ability to create custom variables

### DISCUSSION THREAD

**Discussion by azagthoth1** [CarlGao4/Demucs-Gui#72] [Ideas]
Hello. As far as I understood, the variables when saving files are {model}, {track}, {stem}, and {ext}. Would it be possible for you guys to add more variables or create something to allow us to create variables ourselves? 

I'd really like an {album} variable, so I'd add a whole folder (i.e. an album) and have it wholly separated in my Backing Tracks folder, instead of having many folders like 01 - First Track, 02 - Second Track, you get it.

I don't know if I'm being clear enough, but that's the gist of it. New sets of variables. {artist} or {album artist} would be a nice one too. My music library does not lack metadata, so all of my files have variables like artist, genre, album title, year, etc. I guess this would be a way of putting them to use too. It's all about making things simpler. Just adding album folders and have their stems sorted by album title (or whatever) in the output path, instead of few default options. 

I'd appreciate it.

**Comment by CarlGao4** [comment]
Have implicated this function. Will be released along with 1.3b1

**Reply by CarlGao4** [reply]
1.3 released!
